src/app/Mappings/components/MappingsTable.tsx

```diff
@@ -1,13 +1,119 @@
 import * as React from 'react';
-import { Mapping, MappingType, INetworkMapping, IStorageMapping } from '../types';
+import {
+  Mapping,
+  MappingType,
+  INetworkMapping,
+  IStorageMapping,
+  ICommonMapping,
+  INetworkMappingItem,
+  IStorageMappingItem,
+} from '../types';
+import { Level, LevelItem, Button, Pagination } from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  ICell,
+  sortable,
+  compoundExpand,
+  classNames as classNamesTransform,
+  IRow,
+} from '@patternfly/react-table';
+import { useSortState, usePaginationState, useSelectionState } from '@app/common/hooks';
+import { IVMwareProvider } from '@app/Providers/types';
+import tableStyles from '@patternfly/react-styles/css/components/Table/table';
+import { OutlinedHddIcon, NetworkIcon, DatabaseIcon } from '@patternfly/react-icons';
+import ProviderStatus from '@app/Providers/components/ProvidersTable/ProviderStatus';
+import VMwareProviderActionsDropdown from '@app/Providers/components/ProvidersTable/VMware/VMwareProviderActionsDropdown';
+import VMwareProviderHostsTable from '@app/Providers/components/ProvidersTable/VMware/VMwareProviderHostsTable';
+import MappingsActionsDropdown from './MappingsActionsDropdown';
 
 interface IMappingsTableProps {
   mappings: Mapping[];
+  mappingType: string;
+  toggleAddEditModal: () => void;
 }
 
 const MappingsTable: React.FunctionComponent<IMappingsTableProps> = ({
   mappings,
+  mappingType,
+  toggleAddEditModal,
 }: IMappingsTableProps) => {
+  const getSortValues = (mapping: ICommonMapping) => {
```

Comment:
I think we should use `Mapping` instead of `ICommonMapping` here. `Mapping` is a union of the two specific mapping interfaces, so a mapping with network-specific or storage-specific properties is a valid `Mapping` but not a valid `ICommonMapping` (which is just what those two interfaces extend).